src/gluonts/torch/model/mqf2/module.py

```diff
@@ -0,0 +1,258 @@
+            feat_static_real=feat_static_real,
+            past_time_feat=past_time_feat,
+            past_target=past_target,
+            past_observed_values=past_observed_values,
+            future_time_feat=future_time_feat,
+            future_target=future_target,
+        )
+
+        hidden_state = hidden_state[:, : self.context_length]
+
+        return hidden_state, scale
+
+    @torch.jit.ignore
+    def output_distribution(
+        self,
+        picnn: SequentialNet,
+        hidden_state: torch.Tensor,
+        scale: Optional[torch.Tensor] = None,
+        inference: bool = False,
+    ) -> torch.distributions.Distribution:
+        """
+        Returns the MQF2Distribution instance
+
+        Parameters
+        ----------
+        picnn
+            A SequentialNet instance of a
+            partially input convex neural network (picnn)
+        hidden_state
+            RNN hidden state (batch_size, context_length, hidden_size)
+        scale
+            scaling of the data (batch_size, 1)
+        inference
+            If True, pass only the last hidden state
+            to the forecaster for prediction
+            Otherwise, pass all the hidden states to train the forecaster
+
+        Returns
+        -------
+        MQF2Distribution instance
+            MQF2 parametrized by hidden_state
+        """
+
+        if inference:
+            hidden_state = hidden_state[:, -1]
+
+        return self.distr_output.distribution(picnn, hidden_state, scale=scale)
+
+    def forward(
+        self,
+        feat_static_cat: torch.Tensor,
+        feat_static_real: torch.Tensor,
+        past_time_feat: torch.Tensor,
+        past_target: torch.Tensor,
+        past_observed_values: torch.Tensor,
+        future_time_feat: torch.Tensor,
+        num_parallel_samples: Optional[int] = None,
+    ) -> torch.Tensor:
+        """
+        Generates the predicted sample paths
+
+        Parameters
+        ----------
+        feat_static_cat
+            Static categorial features (batch_size, num_feat_static_cat)
+        feat_static_real
+            Static real-valued features (batch_size, num_feat_static_real)
+        past_time_feat
+            Past time features (batch_size, history_length, num_features)
+        past_target
+            Past target values (batch_size, history_length)
+        past_observed_values
+            Indicator whether or not the values were observed
+            (batch_size, history_length)
+        future_time_feat
+            Future time features (batch_size, prediction_length, num_features)
+        num_parallel_samples
+            Number of parallel sample paths generated for each time series
+
+        Returns
+        -------
+        sample_paths
+            Sample paths (batch_size, num_parallel_samples, prediction_length)
+        """
+
+        if num_parallel_samples is None:
+            num_parallel_samples = self.num_parallel_samples
+
+        picnn = self.picnn
```

Comment:
This is not needed, following the previous comments. I think the two objects will point to the same memory slot so if you don't do a deep copy you are not gaining anything eitherway (you can fact check me on that).